web_src/js/features/notification.js

```diff
@@ -49,11 +49,28 @@ export function initNotificationCount() {
     return;
   }
 
+  const poller = () => {
+    const pollerFn = (timeout, lastCount) => {
+      if (timeout <= 0) {
+        return;
+      }
+      setTimeout(() => {
+        const _promise = updateNotificationCountWithCallback(pollerFn, timeout, lastCount);
+      }, timeout);
+    };
+
+    pollerFn(notificationSettings.MinTimeout, notificationCount.text());
+  };
+
   if (notificationSettings.EventSourceUpdateTime > 0 && !!window.EventSource && window.SharedWorker) {
     // Try to connect to the event source via the shared worker first
     const worker = new SharedWorker(`${__webpack_public_path__}js/eventsource.sharedworker.js`, 'notification-worker');
-    worker.addEventListener('error', (event) => {
-      console.error(event);
+    worker.addEventListener('error', (error) => {
+      if (error.message && error.message === 'ReferenceError: EventSource is not defined') {
```

Comment:
I'm absolutely **not** a fan of comparing against error messages, and even less of doing some strange behavior in that case.
At least add a comment, if comparing against a specific error is impossible.

Reply:
Agree, it's not necessary to use the error message if the error can be detected by a stable method.

Reply:
What is the specific error? Where is the specific error defined in the specs?

Reply:
Comment https://github.com/go-gitea/gitea/pull/20575#issuecomment-1200702977